pkg/kubelet/volumemanager/populator/desired_state_of_world_populator.go

```diff
@@ -384,22 +383,13 @@ func (dswp *desiredStateOfWorldPopulator) checkVolumeFSResize(
 		// or online resize in subsequent loop(after we confirm it has been mounted).
 		return
 	}
-	fsVolume, err := util.CheckVolumeModeFilesystem(volumeSpec)
-	if err != nil {
-		klog.Errorf("Check volume mode failed for volume %s(OuterVolumeSpecName %s): %v",
-			uniqueVolumeName, podVolume.Name, err)
-		return
-	}
-	if !fsVolume {
-		klog.V(5).Infof("Block mode volume needn't to check file system resize request")
-		return
-	}
 	if processedVolumesForFSResize.Has(string(uniqueVolumeName)) {
 		// File system resize operation is a global operation for volume,
 		// so we only need to check it once if more than one pod use it.
 		return
 	}
-	if mountedReadOnlyByPod(podVolume, pod) {
+	// We need to be consistent with what we check.
+	if volumeSpec.ReadOnly {
```

Comment:
Should we pass readonly down to the plugin and let the plugin decide if resize should be skipped?  Or actually let it fail?  What does it mean to resize a read only volume?

Reply:
yeah that is a good idea and ties into discussion we had this yesterday in CSI spec meeting.  Currently `NodeExpandVolumeRequest` can't carry over that information and @bswartz wanted readonly volumes to be expandable on nodes too. In his case - all the plugin would do is, rescan iscsi lun. 

I think, we should not be making that change right now in this PR.

Reply:
I'm wondering if volumeSpec.ReadOnly is sufficient enough, or do we also need to be checking container readonly too?

Reply:
The other question I have is more philosophical.  If a user requests expansion on a read-only volume, should we actually ignore it and return success?  We didn't actually expand the filesystem to what they requested.

Reply:
For CSI - in the long run we want the decision to be taken by the plugin(SP) and not the CO. For in-tree implementations we currently support, doing resizing of readonly volumes with filesystem on it - could result in filesystem corruption.

> I'm wondering if volumeSpec.ReadOnly is sufficient enough, or do we also need to be checking container readonly too?

The container readonly is handled by CRI and is an option for bind mount. I think checking `volumeSpec.Readonly` should be sufficient for our use case.

Reply:
Shouldn't the resizing attempt fail if the volume is attached/mounted as read only?

Reply:
In many cases - expanding the filesystem does not care how volume is mounted.  In many cases expanding(resize2fs/xfsgrow) the volume just takes device path (not the mount path).

Reply:
So the PV.readOnly field is only used for determining if volumes should be attached readonly: https://github.com/kubernetes/kubernetes/issues/70505

Is that the only readOnly setting you want to skip here?

Reply:
Yes - so attached as `ReadOnly` and mounted(NodePublished) as `ReadOnly` are two different things. Reading the linked issue, it seems like we are not setting `PersistentVolume.spec.CSIPersistentVolumeSource.readOnly` anywhere but once we do - this code should be updated to respect that and *not* resize a readonly attached volume.

Again this is with the caveat that - once we have option to pass `ReadOnly` to `NodeExpandRequest` - we will have to update the call to pass it as a flag and let SP decide, how best to handle it.